### DISCUSSION TITLE
잔여백신 예약시도중 화면 질문드립니다.

### DISCUSSION THREAD

**Discussion by RoyCoppedge** [kimytsc/covid-rest-vaccine-macro#25] [Q&A] [answered]
![z](https://user-images.githubusercontent.com/88667060/128713132-ba1a562a-a688-40b0-8dbd-95960588d809.PNG)

안녕하세요.
주신대로 소스를 넣었는데, 현재 시각 22시 23분 기준으로는 보여드린 사진대로 예약시도중 화면입니다.
예약시도 상태에서 "현재 잔여백신이 없습니다" 라고 뜨고 있습니다. 
시간이 22시라 병원들 서칭을 안하는것일까요? 
아까 회사에서 잠깐 해봤을때는 근처 병원들이 돌아가면서 서칭되고 있었거든요.

**Comment by kimytsc** [comment, marked as answer]
리스트 수가 잘 뜨면 정상입니다.

그리고 방금 테스트해본 결과, 현재 병원들이 모두 종료된 상태라 API에서 잔여백신 수를 0개로 내려주고 있는 것으로 파악됩니다. 아침에 다시 시도해보시면 될 것 같습니다.

**Reply by RoyCoppedge** [reply]
답변 감사합니다.
그냥 지금 계속 켜둔 상태로 내일 아침까지 놔둬도 무방할까요?
계속 초마다 갱신되는것보면 병원 오픈과 동시에 사이클 돌아갈것 같긴한데..
궁금해서 추가 질문 드립니다.

**Reply by kimytsc** [reply]
차단의 가능성을 염두해둔다면 계속 켜두기보다 아침에 다시 켜는게 좋지 않을까 생각됩니다.

**Reply by RoyCoppedge** [reply]
안녕하세요.
매크로 방식이 어떻게 돌아가는 방식인지 잘 모르겠어서 질문드립니다.
지도 설정후 좌표 안에 있는 병원들을 하나씩 클릭해서 확인하는 형식인지 아니면 수량이 나오는 곳을 서칭해서 잡아주는 형식인지 궁금합니다. 병원수를 많이 잡을수록 확률이 올라가는것인가요? 병원수를 적게 보고 돌리는것이 좋을까요?

**Reply by kimytsc** [reply]
매크로 방식은 네이버와 카카오가 약간 다르게 동작합니다.

네이버는 아래와 같은 순서로 동작합니다.
1. 지도 API(graphql)에서 100개(start ~ start + 100)의 목록을 받음
2. 목록에서 잔여백신의 총 수량(vaccineQuantity.totalQuantity) 파악
3. 총 수량이 0보다 크다면 상세 백신(vaccineQuantity.list)에서 어떤 백신이 남았는지 파악
4. 선택한 백신의 수량이 남았다면 신청
5. 1~4가 끝나면, 리스트가 100개 이상일 경우, 다음 100개를 가져오고 마지막 리스트라면 처음 리스트를 가져오도록 반복

카카오는 아래와 같은 순서로 동작합니다.
1. 지도 API(left_count_by_coords)에서 전체 목록을 받음
2. 목록에서 잔여백신의 총 수량(leftCounts) 파악(네이버와 다르게 상세 백신 정보가 없음, AZ 잔여백신 이슈 발생)
3. 총 수량이 0보다 크다면 해당 병원의 정보(org_code)를 확인
4. 해당 병원 정보의 상세 백신(lefts)에서 어떤 백신이 남았는지 파악
5. 선택한 백신의 수량이 남았다면 신청
6. 1~5가 끝나면 1번부터 다시 반복

그래서 병원수는 적당히(추천은 100~300정도) 설정하시는게 좋을 것 같습니다.

**Reply by RoyCoppedge** [reply]
상세한 답변 정말 감사드립니다.
덕분에 어제 백신 접종 맞았습니다.
좋은 하루 되세요^^
